alpaca_trade_api/rest.py

```diff
@@ -612,29 +612,38 @@ def _data_get(self,
                   symbol_or_symbols: Union[str, List[str]],
                   api_version: str = 'v2',
                   endpoint_base: str = 'stocks',
+                  resp_grouped_by_symbol: bool = None,
                   **kwargs):
         page_token = None
         total_items = 0
         limit = kwargs.get('limit')
+        page_limit = DATA_V2_MAX_LIMIT
+        if endpoint_base == "news":
+            page_limit = NEWS_MAX_LIMIT
         while True:
             actual_limit = None
             if limit:
-                actual_limit = min(int(limit) - total_items, DATA_V2_MAX_LIMIT)
+                actual_limit = min(int(limit) - total_items, page_limit)
                 if actual_limit < 1:
                     break
             data = kwargs
             data['limit'] = actual_limit
             data['page_token'] = page_token
-            if isinstance(symbol_or_symbols, str):
-                path = f'/{endpoint_base}/{symbol_or_symbols}/{endpoint}'
-            else:
-                path = f'/{endpoint_base}/{endpoint}'
+            path = f'/{endpoint_base}'
+            if isinstance(symbol_or_symbols, str) and symbol_or_symbols:
+                path += f'/{symbol_or_symbols}'
+            if endpoint:
+                path += f'/{endpoint}'
+            if isinstance(symbol_or_symbols, list) and symbol_or_symbols:
                 data['symbols'] = ','.join(symbol_or_symbols)
-            if not endpoint_base or not symbol_or_symbols or not endpoint:
-                path = re.sub(r'(\/)\1+', r'\1', path)  # Remove extra slashes
             resp = self.data_get(path, data=data, api_version=api_version)
-            if isinstance(symbol_or_symbols, str) or endpoint == "news":
-                for item in resp.get(endpoint, []) or []:
+            if (
+                resp_grouped_by_symbol is False
+                or isinstance(symbol_or_symbols, str)
+            ):
```

Comment:
[minor] At the end of the day it's the same, but I feel it would be a bit more pythonic to check for None first outside the loop:
```python
if resp_grouped_by_symbol is None:
    resp_grouped_by_symbol = not isinstance(symbol_or_symbols, str)
```
and then simply use `resp_grouped_by_symbol` here as a condition.
```